### PR TITLE
Update view Timestamp logic to fix setAmendmentState problem

### LINKED ISSUE #4303: Motion detail view with amendments: Changing amendment status needs reload
**Describe the bug**
It is possible to change the status of an amendment in the motion detail view, but the change is not displayed. The status switches up, but the old status is still displayed. An update that updates the visulization seems to be missing here.

**How to Reproduce**
1. open a meeting
2. open motions > create a motion with an amendment
3. open the motion in the detail view
4. mark the checkbox 'Show all changes' > try to change the amendment status via the three-dot-menu behind the amendment

**Expected behavior**
Changing the amendmend status via the three-dot-menu should have visual effect. New status should be immediately visible

### PR DESCRIPTION
**Pull request by reiterl**
Resolve #4303 

Remove a strange distinct until changed timestamp logic which leads to failures in the issue.
I don't know, what the reason for this logic is, I think performance problems. So a whole test of the motion detail component is needed.

### CODE REVIEW

**Review by luisa-beerboom — COMMENTED** [review]
The `motion-change-recommendation-controller.service.ts`  there is a method `getChangeRecosOfMotionObservable` that has the exact same code. If this code is broken, that one should be as well. Please check that.
If the change recommendation code is also broken, the entire `viewModelUpdateTimestamp` logic can be removed. If not, Maybe reconsider whether this `distinctUntilChanged` is actually at fault

**Comment by reiterl** [comment]
I worked on fixing the update-timestamp at update, used the changedModels method for it. Test works.

**Comment by reiterl** [comment]
@luisa-beerboom  Please look at the my new solution.

**Review by luisa-beerboom — APPROVED** [review]
You should probably change the title of this PR

**Review by Elblinator — CHANGES_REQUESTED** [review]
This still doen't fix the Issue:
I created a motion with 8 paragraphs 
and added 5 amendments (in paragraph 1,2,3,7 and 8)

- Changing the first state sometimes works
- Sometimes it updates after three amendment's states were changed
- sometimes it updates if I reset all changes amendments to submitted

step to step reproduction:
1. create motion with 200 words and 8 paragraphs (blindtextgenerator)
2. Create 5 amendments (in line 1-2, 3-4, 5, 13-14 and 15), I deleted the first word for each amendment
3. updating to the state to "accepted" or "not decided" (of any amendment) updates immediately, updating to "not decided" does not change the view automatically 

(I called stop-dev and run-dev, the workflows are the default dev ones

**Comment by reiterl** [comment]
@Elblinator showed me the problem on his computer. I build a small change, comparing the state_ids in the distinctUntil part.

**Review by bspekker — APPROVED** [review]
functionality tested

File: client/src/app/site/pages/meetings/pages/motions/services/common/amendment-controller.service/amendment-controller.service.ts [hunk 1]
```diff
@@ -47,7 +47,8 @@ export class AmendmentControllerService {
                 (prev, curr) =>
                     prev?.length === curr?.length &&
                     Math.max(...prev.map(e => e.viewModelUpdateTimestamp)) ===
-                        Math.max(...curr.map(e => e.viewModelUpdateTimestamp))
+                        Math.max(...curr.map(e => e.viewModelUpdateTimestamp)) &&
+                    prev?.map(e => e.state_id).equals(curr?.map(e => e.state_id))
```
[the remark this answers is not in the document]
Comment (bastianjoel) [reply client/src/app/site/pages/meetings/pages/motions/services/common/amendment-controller.service/amendment-controller.service.ts:51]: Also this should be done in `motion-change-recommendation-controller.service.ts` too.

**Pull request merged**